package/CodePoint.roc

```diff
@@ -29,40 +29,34 @@ toU32 = InternalCP.toU32
 ## (that is, it's between `0` and `0x10FFFF`).
 fromU32 : U32 -> Result CodePoint [InvalidCodePoint]
 fromU32 = \u32 ->
-    # Definition: http://www.unicode.org/glossary/#code_point
     if u32 <= 0x10FFFF then
         Ok (fromU32Unchecked u32)
     else
         Err InvalidCodePoint
 
-## Returns false if this is either a [high-surrogate code point](http://www.unicode.org/glossary/#high_surrogate_code_point)
-## or a [low-surrogate code point](http://www.unicode.org/glossary/#high_surrogate_code_point).
-##
-## To check for either of those individually, use [isHighSurrogate] or [isLowSurrogate]
+## Returns false if this is [isHighSurrogate] or [isLowSurrogate]
 isValidScalar : CodePoint -> Bool
 isValidScalar = \codePoint -> !(isHighSurrogate codePoint || isLowSurrogate codePoint)
 
 ## Returns true if this is a [high-surrogate code point](http://www.unicode.org/glossary/#high_surrogate_code_point)
-## (`0xD800` to `0xDBFF`)
+## from U+D800 to U+DBFF
 isHighSurrogate : CodePoint -> Bool
 isHighSurrogate = \codePoint ->
-    u32 = InternalCP.toU32 codePoint
-
-    u32 >= 0xDC00 && u32 <= 0xDFFF
+    u32 = toU32 codePoint
+    u32 >= 0xD800 && u32 <= 0xDBFF
 
 ## Returns true if this is a [low-surrogate code point](https://www.unicode.org/glossary/#low_surrogate_code_point)
-## U+DC00 to U+DFFF
+## from U+DC00 to U+DFFF
 isLowSurrogate : CodePoint -> Bool
 isLowSurrogate = \codePoint ->
-    u32 = InternalCP.toU32 codePoint
-
+    u32 = toU32 codePoint
     u32 >= 0xDC00 && u32 <= 0xDFFF
 
 ## Zig docs: bytes the UTF-8 representation would require
 ## for the given codepoint.
-utf8Len : CodePoint -> Result U64 [InvalidCodePoint]
+utf8Len : CodePoint -> Result U8 [InvalidCodePoint]
```

Comment:
We used `U64` here as it was recently changed from `Nat`. I don't really know myself, but just wondering why you changed it to `U8`?

Reply:
This was one of my doubts and should've asked before. But `utf8Len` and `countUtf8Bytes` return values in the range of 1 to 4.

Reply:
Lets give it a try. It's always easy to cast if needed